action/protocol/poll/consortium.go

```diff
@@ -281,7 +292,7 @@ func genContractReaderFromReadContract(r ReadContract, setting bool) contractRea
 	}
 }
 
-func getContractReaderForGenesisStates(ctx context.Context, sm protocol.StateManager, getBlockHash evm.GetBlockHash) contractReaderFunc {
+func getContractReaderForGenesisStates(ctx context.Context, sm protocol.StateManager) contractReaderFunc {
```

Comment:
it seems it's only used for genesis block, so we can know Shanghai is not enabled, no need to pass in a real func